[16.0][IMP] auth_saml: Allow admin/managers to edit User SAML information

vincent-hatakeyama opened the pull request:
As user in that group can already edit users, so it make sense to allow them to see and edit that information rather than restrict it to admin/system.

aronabencherif committed:
[IMP] auth_saml: Allow admin/managers to edit User SAML information

As user in that group can already edit users, so it make sense to allow
them to see and edit that information rather than restrict it to
admin/system.

PCatinean commented on auth_saml/security/ir.model.access.csv:
```diff
@@ -1,6 +1,5 @@
-access_auth_res_users_saml,auth_res_users_saml,model_res_users_saml,base.group_system,1,1,1,1
+access_auth_res_users_saml,auth_res_users_saml,model_res_users_saml,base.group_erp_manager,1,1,1,1
 access_auth_saml_attribute_mapping,auth_saml_attribute_mapping,model_auth_saml_attribute_mapping,base.group_system,1,1,1,1
-auth_saml_provider_erp_manager_access,auth_saml_provider_erp_manager_access,model_auth_saml_provider,base.group_erp_manager,1,0,0,0
```
I think we might still need this given that res.users.saml has a many2one to auth.saml.provider and one needs to select the provider when making/editing an entry

vincent-hatakeyama commented:
/ocabot merge minor

vincent-hatakeyama committed:
[FIX] auth_saml: update signin method

Updated the signin method to reflect changes in similar method signin
from auth_oauth.
Without the changes, the ORM crashes with
psycopg2.errors.InvalidSavepointSpecification when trying to signin.

Fixes #664

vincent-hatakeyama commented:
/ocabot merge minor
